CONTRIBUTING.md

```diff
@@ -95,8 +95,8 @@ separately.
 
 The docs are published off of three branches:
 
-- `book-v2`: [book.kubebuilder.io](https://book.kubebuilder.io) -- current
-  docs
+- `book-v3`: [book.kubebuilder.io](https://book.kubebuilder.io) -- current docs
+- `book-v2`: [book-v2.kubebuilder.io](https://book.kubebuilder.io) -- legacy docs
```

Comment:
This PR should probably wait until a `book-v2` subdomain is created. @DirectXMan12 how would we go about doing that?